lib/calrom/option_parser.rb

```diff
@@ -252,5 +260,15 @@ def build_date_range(range_type, year, month, day, another_day=nil)
 
       range
     end
+
+    def last_month
```

Comment:
1. Methods computing details of date ranges are not welcome in the `OptionParser` class. Please extract this logic to a new subclass of `DateRange`.
2. The code is unnecessarily complicated. Don't touch the `Time` class (which is - for a good reason - so far not used a single time in `calrom` codebase), rewrite the code using the [`Date`](https://ruby-doc.org/stdlib-2.6.5/libdoc/date/rdoc/Date.html) class only. The `Date` class has several methods useful for date arithmetics, make sure to learn about them and use them.

Reply:
Can I abstract the logic to DateRange like this...?
```ruby
  class LastMonth < DateRange
    def initialize(year, month)
      super Date.new(year, month - 1)
    end

    def to_s
      first.year.to_s
    end

    def each_month
      return to_enum(:each_month) unless block_given?

      1.upto(12) {|month| yield Month.new(first.year, month) }
    end
  end

  class NextMonth < DateRange
    def initialize(year, month)
      super Date.new(year, month + 1, -1)
    end

    def to_s
      first.year.to_s
    end

    def each_month
      return to_enum(:each_month) unless block_given?

      1.upto(12) {|month| yield Month.new(first.year, month) }
    end
  end
``` 

And... in the option_parse I should call the class like this... ?
```ruby
        when :'3'
          last_month = LastMonth.new(Date.today.year, Date.today.month)
          next_month = NextMonth.new(Date.today.year, Date.today.month)
          DateRange.new(last_month, next_month)
        ...
```
It's correctly ?

Reply:
No. Something along the lines of

```ruby
class ThreeMonths < DateRange
  def initialize(date)
    super first_day_of_last_month(date), last_day_of_next_month(date)
  end

  private

  # TODO: implement the missing methods here
end
```

and then in `OptionParser`

```ruby
# ...
when :three_months
  ThreeMonths.new(day)
# ...
```